### ISSUE TITLE
Creating chatbot in other language

### ISSUE THREAD

**Issue by Wickky** [facebookresearch/ParlAI#2648]
Are we able to use this for building a chatbot for another language?

**Comment by stephenroller** [comment]
Sure, the tools are language agnostic. Unfortunately all of our current data sets are in English. We would greatly welcome dialogue datasets in other languages.

**Comment by Wickky** [comment]
Thank you for answering my question. For training for other language, is it fine to only change the dataset in other language? What else do we need?

**Comment by stephenroller** [comment]
It’s hard to say, the situation you’ve described is too vague. Ostensibly yes, but maybe not depending on your situation.

**Comment by jsedoc** [comment]
The main issue you will face is the tokenization. Otherwise, everything should work. Languages like Spanish work almost entirely out of the box whereas Chinese will take some work.

**Comment by stephenroller** [comment]
We just added c3, which is a Chinese reading comprehension dataset. It's not dialogue, but it's a start! #2665.

**Comment by Wickky** [comment]
@stephenroller  @jsedoc  
Thank you for your suggestions.
For the tokenization case, where can I change it? 
Suppose I have a tokenization model for my own language, where can I update the code to work?

**Comment by stephenroller** [comment]
If you want to use a BPE tokenizer, you can follow HuggingFace's instructions to [train your own](https://github.com/huggingface/tokenizers). If you have a premade tokenizer, you can add say `wickky_tokenize` to this class:

https://github.com/facebookresearch/ParlAI/blob/d191b8063a8b8500737f73b658188d393cc96211/parlai/core/dict.py#L391-L402

```python
def wickky_tokenize(self, text):
     return text.split("x")
```

And then use rest of parlai with the `--dict-tokenizer wickky` argument.

**Comment by Wickky** [comment]
@stephenroller  Thank you so much! It works!

**Comment by Wickky** [comment]
@stephenroller  I have one more question sir. When we predict say in interactive mode, does it use the tokenizer we passed while training for the user input message ?

**Comment by stephenroller** [comment]
Yes.

**Comment by Erdene-Ochir0417** [comment]
Hello @stephenroller
I have one question. How can I use sentence piece tokenizer?

**Comment by Hilal-Urun** [comment]
Hi @stephenroller  when I tried to add tokenizer and fine-tuning bart model with Turkish Wiki data it gives error. like ;
`AttributeError: 'bool' object has no attribute 'cumsum'`   or torch size error. I fixed n-positions too but I can not handle that. For example it runs last night on vast.ai but it stops because of the credit, then i run today again but it does not works, i could not understant :D
